src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +432,95 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
```

Comment:
Yes.. using async was a mistake. In combination with not removing the PollFunction in the constructor was causing weird errors. I have now replaced with Task.Run. I haven't yet removed the Delay - will do so shortly